### PR TITLE
fix!: stop provide Web API polyfills in components package

### PR DESCRIPTION
**Pull request by dozortsev**
## 🖼 Context

Historically, sometimes we have been providing Web API polyfills, but we have be doing that inconsistently (some have polyfills some not). In this PR we decided to align our components on that, and we officially drop support of Web API polyfills. 

This doesn't make library worse, in fact vice-versa, we won't ship in most case redundant code (as percentage of browsers that need them is quite small), less fuss for maintainers. It's way more convenient to provide polyfills at consumers side and there are plenty of great tools that would do that in best way (lazy loading) polyfill-library, Polyfill.io, NextJS etc.

## 🚀 Changes

- **Breaking changes:** Remove existing polyfills for some components
- Update docs, added info about requireded Web API polyfills: [table of Neptune API that require polyfills at usage docs](https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/about/Setup#polyfills-for-web-browser-api) and section about polyfills at components pages (e.g [Upload](https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/components/inputs/Upload#polyfills), [Image](https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/components/content/Image#polyfills) etc.)

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * Common Web API Polyfills
+ */
+import './common/polyfills/intl';
```
Comment (dozortsev) [inline packages/components/src/index.ts:4]: note: I moved import here as `Intl.*` API used not only by `Provider` component, atm afaik we also we it in [locale utils](https://github.com/transferwise/neptune-web/blob/main/packages/components/src/common/locale/index.js#L47-L94)

File: packages/docs/pages/about/Setup.mdx [hunk 2]
```diff
@@ -190,6 +190,94 @@ import '@transferwise/neptune-css/dist/css/neptune-addons.css';
 
 The components are exposed in two formats: UMD and ES. We also provide polyfilled and non-polyfilled versions.
 
+<h3 id="polyfills-for-web-browser-api">Polyfills for Web / Browser API</h3>
+
```
Comment (dozortsev) [inline packages/docs/pages/about/Setup.mdx:194]: https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/about/Setup#polyfills-for-web-browser-api

File: packages/components/src/dateInput/DateInput.js [hunk 3]
```diff
@@ -3,7 +3,6 @@ import { useIntl } from 'react-intl';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 
-import '../common/polyfills/closest';
```
Comment (fergusjordantw) [inline packages/components/src/dateInput/DateInput.js:6]: I'm guessing by the change in `packages/docs/pages/_app.js` these docs are deployed on a crab app which comes with polyfills, which means this is problematic for most consumers: 

image

Comment (dozortsev) [reply packages/components/src/dateInput/DateInput.js:6]: @fergusjordantw thanks for noticing this 🙇 

the docs deployed as NextJS static build to Github Pages,

re. CRAB, it should enable polyfills required by Neptune web https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/about/Setup#polyfills-for-web-browser-api

Comment (fergusjordantw) [reply packages/components/src/dateInput/DateInput.js:6]: Looks good as long as we can get that crab PR merged pending a test in success-screens that the latest version of polyfill-library doesn't have the same memory leak it used to that prevented us from using later versions. Will try and look later today

File: packages/docs/pages/about/Setup.mdx [hunk 4]
```diff
@@ -62,6 +62,94 @@ For information on what classes are available, see the <DocLink href="/styles/co
 
 By default, apps come polyfilled. If you would like to use an unpolyfilled version because your app provides its own polyfills, see [Advanced Components Usage](#advanced-components-usage) below.
 
+<h2 id="polyfills-for-web-browser-api">Polyfills for Web / Browser API</h2>
+
+We don't provide polyfills for <a href="https://developer.mozilla.org/en-US/docs/Web/API">Web API</a>, please provide it yourself in whichever way it best works for you.
+
+Table of API that require some polyfills.
```
Comment (helendurranttw) [inline packages/docs/pages/about/Setup.mdx:69]:
```suggestion
<h2 id="polyfills-for-web-browser-api">Polyfills for IE</h2>

If you need to support Internet Explorer, you'll need some <a href="https://developer.mozilla.org/en-US/docs/Web/API">Web API</a> polyfills. An easy way to include them is using <a href="https://polyfill.io/v3/">Polyfill.io</a>

Components and their required polyfills
```

File: packages/docs/pages/about/Setup.mdx [hunk 5]
```diff
@@ -62,6 +62,94 @@ For information on what classes are available, see the <DocLink href="/styles/co
...
+  <thead>
+    <tr>
+      <th>Item</th>
+      <th>Require polyfill(s) for</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>
+        <DocLink href="components/context/Provider">Provider</DocLink> component
```
Comment (helendurranttw) [inline packages/docs/pages/about/Setup.mdx:81]:
```suggestion
        <DocLink href="components/context/Provider">Provider</DocLink>
```

File: packages/docs/pages/about/Setup.mdx [hunk 6]
```diff
@@ -62,6 +62,94 @@ For information on what classes are available, see the <DocLink href="/styles/co
...
+            <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale">
+              <code>Intl.Locale</code>
+            </a>
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <DocLink href="components/content/Image">Image</DocLink> component
```
Comment (helendurranttw) [inline packages/docs/pages/about/Setup.mdx:95]:
```suggestion
        <DocLink href="components/content/Image">Image</DocLink>
```

File: packages/docs/pages/about/Setup.mdx [hunk 7]
```diff
@@ -62,6 +62,94 @@ For information on what classes are available, see the <DocLink href="/styles/co
...
+            <a href="https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/IntersectionObserver">
+              <code>IntersectionObserver</code>
+            </a>
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <DocLink href="components/inputs/DateInput">DateInput</DocLink> component
```
Comment (helendurranttw) [inline packages/docs/pages/about/Setup.mdx:109]:
```suggestion
        <DocLink href="components/inputs/DateInput">DateInput</DocLink>
```

File: packages/docs/pages/about/Setup.mdx [hunk 8]
```diff
@@ -62,6 +62,94 @@ For information on what classes are available, see the <DocLink href="/styles/co
...
+            <a href="https://developer.mozilla.org/en-US/docs/Web/API/Element/closest">
+              <code>Element#closest</code>
+            </a>
+          </li>
+        </ul>
+      </td>
+    </tr>
+    <tr>
+      <td>
+        <DocLink href="components/inputs/Upload">Upload</DocLink> component
```
Comment (helendurranttw) [inline packages/docs/pages/about/Setup.mdx:123]:
```suggestion
        <DocLink href="components/inputs/Upload">Upload</DocLink>
```

File: packages/docs/pages/about/BrowserSupport.mdx [hunk 9]
```diff
@@ -1,4 +1,8 @@
-The table below shows what we currently support.
+import { DocLink } from '../../utils/pageUtils';
+
+Table of browsers that we provide polyfills for standard JS API and syntax features using Babel.
+
+Note: <DocLink href="about/Setup#polyfills-for-web-browser-api">You might need add some polyfills for Web API</DocLink>.
```
Comment (helendurranttw) [inline packages/docs/pages/about/BrowserSupport.mdx:5]:
```suggestion
Note: If you want to support Internet Explorer, you'll need a few <DocLink href="about/Setup#polyfills-for-web-browser-api">extra polyfills</DocLink>.
```

File: packages/docs/pages/about/Setup.mdx (hunk 4 above)
Comment (dozortsev) [reply packages/docs/pages/about/Setup.mdx:69]: polyfills are not only for IE, e.g polyfill for `Intl.Locale` is necessary for some previous versions of Safari, Firefox

polyfills necessary not only for components, some util functions too

File: packages/docs/pages/about/Setup.mdx (hunk 5 above)
Comment (dozortsev) [reply packages/docs/pages/about/Setup.mdx:81]: okey, I remove word "component" from here but again polyfills necessary not only for components

https://transferwise.github.io/neptune-web/branch/fix-intl.locale-polyfill/about/Setup#polyfills-for-web-browser-api

File: packages/docs/pages/about/BrowserSupport.mdx (hunk 9 above)
Comment (dozortsev) [reply packages/docs/pages/about/BrowserSupport.mdx:5]: why deleting first sentence?

we do provide polyfills for Standard JS API based on list of browsers that are in below table via [`browserslist`](https://github.com/transferwise/neptune-web/blob/main/package.json#L96-L104) so we kinda saying about that

**Review by matebek-tw — APPROVED** [review]

**Pull request merged**